app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -59,28 +53,30 @@ public void onEntryNameSubmitted() {
     }
 
     @Override
-    public void setContestName(String contestName) {
-        contestantWelcomeTextView.setText(getResources().getString(R.string.contestant_welcome_message, contestName));
+    public void showWelcomeMessage() {
+        setActionBarTitle(getResources().getString(R.string.contestant_welcome_message));
     }
 
     @Override
-    public void showEntryNameButton() {
+    public void enableEntryNameButton() {
+        entryNameNextButton.setEnabled(true);
         entryNameNextButton.setVisibility(android.view.View.VISIBLE);
     }
 
     @Override
-    public void hideEntryNameButton() {
+    public void disableEntryNameButton() {
+        entryNameNextButton.setEnabled(false);
         entryNameNextButton.setVisibility(android.view.View.GONE);
     }
 
     @Override
-    public void enableEntryNameButton() {
-        entryNameNextButton.setEnabled(true);
+    public void showIcon() {
+        entryNameIcon.setVisibility(android.view.View.VISIBLE);
```

Comment:
Bit of a nit, but let's static import the Visibles/Gones to clean the code up a bit.